src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/fix/data-broker-profiles/removal-under-maintenance/RemovalUnderMaintenanceView.tsx

```diff
@@ -0,0 +1,403 @@
+import { TelemetryButton } from "../../../../../../../../../components/client/TelemetryButton";
+import { CONST_URL_SUMO_MANUAL_REMOVAL } from "../../../../../../../../../../constants";
+import React, { useEffect, useState } from "react";
+import {
+  BackArrow,
+  ClockIcon,
+} from "../../../../../../../../../components/server/Icons";
+import { LatestOnerepScanData } from "../../../../../../../../../../db/tables/onerep_scans";
+import { ExposureCardDataClassLayout } from "../../../../../../../../../components/client/exposure_card/ExposureCardDataClass";
+import confetti from "canvas-confetti";
+import { OnerepScanResultRow } from "knex/types/tables";
+import { toast } from "react-toastify";
+
+export type Props = {
+  data: LatestOnerepScanData;
+  stepDeterminationData: StepDeterminationData;
+  subscriberEmails: string[];
+};
+
+export const RemovalUnderMaintenanceView = (props: Props) => {
+  const l10n = useL10n();
+  const [detailedRemovalGuide, setDetailedRemovalGuide] = useState(false);
+  const [firstScanResultNotResolved, setFirstScanResultNotResolved] = useState(
+    props.data.results[0],
+  );
+  const [isLoadingNextDataBroker, setIsLoadingNextDataBroker] = useState(false);
+  const [fadeState, setFadeState] = useState<"fadeIn" | "fadeOut" | "">("");
+
+  const nextGuidedStep = getNextGuidedStep(
+    props.stepDeterminationData,
+    "DataBrokerManualRemoval",
+  );
+
+  const [nextScanResultNotResolved, setNextScanResultNotResolved] = useState<
+    OnerepScanResultRow | undefined
+  >();
+
+  const updateDataBrokerWithTimeout = () => {
+    setIsLoadingNextDataBroker(true);
+    setFadeState("fadeOut");
+
+    setTimeout(() => {
+      setIsLoadingNextDataBroker(false);
+      void confetti();
+      setFadeState("fadeIn"); // trigger the fade-in for the next scan result
+    }, 2000);
+  };
+
+  async function handleManualRemovalChange() {
+    try {
+      const response = await fetch(
+        `/api/v1/user/scan-result/${firstScanResultNotResolved.onerep_scan_result_id}/resolution`,
+        {
+          method: "POST",
+          credentials: "same-origin",
+        },
+      );
+
+      if (!response.ok) {
+        toast.error("Could not resolve data broker.");
+        console.error(
+          "Could not update next data broker with removal under maintenance status.",
+        );
+        return;
+      }
+
+      // Mark the current scan result as manually resolved
+      firstScanResultNotResolved.manually_resolved = true;
+
+      const nextUnresolved = props.data.results.find(
+        (scanResult) => !scanResult.manually_resolved,
+      );
+
+      // Redirect if no unresolved scan result remains
+      if (!nextUnresolved) {
+        window.location.href = nextGuidedStep.href;
+      } else {
+        setNextScanResultNotResolved(nextUnresolved);
+      }
+
+      // Trigger the timeout update
+      updateDataBrokerWithTimeout();
```

Comment:
Calling this function here will only set the loading state after the actual fetch request to resolve the result is done. In that time period a user could still click the button below again and trigger this function. We can do two thing to prevent this:
- check if `isLoadingNextDataBroker` is true at the beginning of `handleManualRemovalChange` and return early if that is the case
- and set call `setIsLoadingNextDataBroker(true)` directly after